### ISSUE TITLE
Incorrect exit() executed when leaving composite state from choice pseudostate

### ISSUE THREAD

**Issue by joeloyj** [steelbreeze/state.js#27]
Example:

```
var state = require('state.js');

var model = new state.StateMachine('model');
var initial = new state.PseudoState('initial', model, state.PseudoStateKind.Initial);
var a = new state.State('a', model);
var b = new state.State('b', model);
var aa = new state.State('aa', a);
var aChoice = new state.PseudoState('aChoice', a, state.PseudoStateKind.Choice);

initial.to(aa);
aa.to(aChoice);
aChoice.to(b).when(function() { return true });

var instance = new state.StateMachineInstance('instance');
state.setConsole(console);
state.initialise(model, instance);
```

Running the above code produces:
```
initialise model
initialise instance
instance enter model
instance enter model.default
instance enter model.default.initial
instance leave model.default.initial
instance enter model.default.a
instance enter model.default.a.default
instance enter model.default.a.default.aa
instance leave model.default.a.default.aa
instance enter model.default.a.default.aChoice
instance leave model.default.a.default.aa
instance leave model.default.a.default
instance leave model.default.a
instance enter model.default.b
```

Note that `instance leave model.default.a.default.aa` is executed twice. The second occurrence should be `instance leave model.default.a.default.aChoice`.

**Comment by mesmo** [comment]
Thanks Joel,

Investigating...

**Comment by mesmo** [comment]
Hi Joel, you've uncovered a bug that will cause what you're seeing whenever you leave a _composite state_ via a transition whose source is a _pseudo state_. The code uses the _last known state_ in the _active state configuration_ (which is always a state as its used for history purposes as well) as the basis for what child states to leave.

In your example, when bootstrapping the transition from ```aChoice``` to ```b``` it determines that it needs to exit ```a``` and then cascades the leave operation using the last know state.

I will investigate potential fixes, but this may take a little time - in the mean time, if

**Comment by mesmo** [comment]
A fix is in place on the master branch; I just need to perform a little more testing before I release.

**Comment by mesmo** [comment]
Unfortunately, the approach I was looking at will not work; I have another angle of attach. This requires that I extend and expand the _active state configuration_ to contain both the _last known state_ and the _current active vertex_.

**Comment by mesmo** [comment]
Fixed by release v5.11.0
